src/scripts/components/UploadListItem.js

```diff
@@ -289,12 +290,14 @@ class UploadListItem extends Component<Props, Object> {
 
   onPublishVideo (e: Object) {
     e.preventDefault()
+    const videoId = this.state.id
```

Comment:
If we have the `video` in props do we need it in `state`? 

If we want the id can we just check `this.props.video.get('id')`?

Reply:
which is the problem to have also the `video id` in the `state`?

Reply:
Two issues:

It is duplication of data. If we have it in two places which is our source of truth (it should be props) and how do we guarantee they don’t get out of sync?

Setting state in addition to props can easily lead us to unnecessary rerenders of the component, which is bad for performance. We should avoid putting things in state that are already in props because it is unnecessary and avoids these problems